flashinfer/cute_dsl/sparse/blk128/pipeline.py

```
@@ -0,0 +1,440 @@
+        #     [Int32],
+        # )
+
+    def __extract_mlir_values__(self):
+        phase_index = self._phase_index
+        return [phase_index.ir_value()]
+
+    def __new_from_mlir_values__(self, values):
+        return PipelineStateSimple(self.stages, Int32(values[0]))
+
+
+def make_pipeline_state(type: PipelineUserType, stages: int):
```

Ruff (0.15.15)

[error] Function argument `type` is shadowing a Python builtin

(A002)